docs/kilnjs.md

```diff
@@ -0,0 +1,220 @@
+---
+id: kilnjs
+title: Kilnjs
+sidebar_label: Kilnjs
+---
+
+---
+
+Kilnjs is an optional Javascript file that transforms the schema.yaml file into a dynamic Javascript Object that affects how Kiln interacts with components that use the schema.  With it, events can be attached to the inputs that are used by kiln to edit components.  These events can be used to hide or reveal different fields in the schema, update the values of those fields, make api calls, subscribe to Vuex mutations, and just about any other action allowed by JavaScript.
+
+Using Kilnjs, you can remove the logic from the schema.yaml file and move it into JavaScript, keeping the schema file for presentational information.
+
+To take advantage of the power that Kilnjs provides you need to create a file called kiln.js within a component folder, in the same location where you place the component's schema, model, and client files. At a minimum, it should look like this.
+```
+'use strict';
+
+module.exports = (schema) => {
+
+  return schema;
+};
+```
+A function that recieves a parameter called schema that also returns the schema.  The schema contains a JSON representation of the schema.yaml file and can be manipulated before returning.  Values can be changed, added, or deleted directly on the JSON object without any other outside code.  Anything you can do to a JSON object can be done to the schema.
+
+```
+schema['_groups'].settings['_placeholder'].height = '200px';
+```
+
+However, note that the JSON that is returned still needs to conform to the structure needed by Kiln.  Deleting properties or adding properties that Kiln doesn't recognize can and will result in errors.
+
+
+---
+
+## KilnInput
+
+The real power of Kilnjs comes from the KilnInput object, which can be used to make the fields within a schema truly dynamic.  Using KilnInput you can add events to the different form inputs as detailed on the [Form Inputs page](input).  You can also subscribe to Vuex actions as described below.
+
+To make a schema field a KilnInput, you set it to a new instance of KilnInput, passing it to the schema and the name of the field.
+
+```
+const KilnInput = window.kiln.kilnInput;
+
+module.exports = (schema) => {
+  schema.title = new KilnInput(schema, 'title');
+
+  return schema;
+};
+```
+
+---
+
+## Events
+
+You can attach events to the schema inputs.  The events you can attach vary depending upon the type input. The different events are detailed on the [Form Inputs page](input).
+
+* ***on(event, callbackFunction)*** - pass it the event as a string, along with a callback function that is run after the event happens
+
+```
+  schema.enableSocialButtons.on('input', (val) => {
+    if(val) {
```

Comment:
```suggestion
    if (val) {
```
Also, remember to add the language to the code snippet for styles and format

Reply:
👍